assignments/callbacks.js

```diff
@@ -2,29 +2,117 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
 
 function firstItem(arr, cb) {
   // firstItem passes the first item of the given array to the callback function.
+  
+  const item =cb(arr);
+  console.log(item);
 }
 
+function printItem(items){
+return items[0];;
+}
+
+firstItem(items,printItem)
+
+
+
+
+
 function getLength(arr, cb) {
   // getLength passes the length of the array into the callback.
+  
+  const item = cb(arr);
+  console.log(item);
+}
+
+
+function printLength(items){
+return items.length;;
 }
 
+firstItem(items,printLength)
+
+
+
+
 function last(arr, cb) {
-  // last passes the last item of the array into the callback.
+  // last passes the last item of the array into the callback
+  
+  const item =cb(arr);
+  console.log(item);
 }
 
+function printItem(items){
+return items[items.length-1];;
+}
+
+last(items,printItem)
+
+
+
+
+
 function sumNums(x, y, cb) {
   // sumNums adds two numbers (x, y) and passes the result to the callback.
+  const sum = cb(x,y);
+  console.log (sum);
 }
 
+function add(x,y){
+	return x+y;
+	]
+	
+sumNums(2,1,add);
+
 function multiplyNums(x, y, cb) {
   // multiplyNums multiplies two numbers and passes the result to the callback.
+
+  const mult = cb(x,y);
+  console.log (mult);
+}
+
+
+
+function mult(x,y){
+	return x*y;
 }
+	
+multiplyNums(2,5,mult);
 
 function contains(item, list, cb) {
   // contains checks if an item is present inside of the given array/list.
   // Pass true to the callback if it is, otherwise pass false.
+  let vlaue = cb(item,list);
+  console.log(value);
+  
 }
 
+function contains(item, list, cb) {
```

Comment:
For this one, your code gives the correct answer.

However, the implementation is a little inside out for what is being asked.

We want `contains` to be doing all of the work here, and to pass either `true` or `false` into the callback function, whatever the callback is.

So your code could be rearranged to fit the requirements by doing this:
```
function contains(item, list, cb) {
  let i = list.indexOf(item);
  if(i < 0) {
    cb(false);
  }
  else {
    cb(true);
  }
}
```
I really like how you're thinking about these problems, I don't want you to think you're doing them wrong. These problems are really abstract and it's kind of hard to tell what you're supposed to be doing, so you're doing great.